File: Servidor/SilentManagerApiServer/BusinessLogic/Dtos/InviteDto.cs
using SilentManager.API.Models.DB;
using SilentManagerApiServer.BusinessLogic.Dtos;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SilentManager.API.BusinessLogic.Dtos
{
    public class InviteDto
    {
        public int UserId { get; set; }
        public int EventId { get; set; }
        public int State { get; set; }
    }

    public class InviteOutDto
    {
        public UserOutDto User { get; set; }
        public CompletedEvent Event { get; set; }
        public int? State { get; set; }
    }

    public class InviteInDto
    {
        public InviteDto Invite { get; set; }
        public TokenInDto Token { get; set; }
    }
}

File: Aplicacao_cliente/SilentManager/app/src/main/java/com/silent_manager/g29/silent_manager/data_layer/models/dtos/InviteDto.kt
package com.silent_manager.g29.silent_manager.data_layer.models.dtos

import com.silent_manager.g29.silent_manager.data_layer.models.Invite

data class InviteDto(
    val userId: Int?,
    val eventId: Int?,
    val state: Int?
) {
    companion object {
        fun convertToDto(invite: Invite): InviteDto {
            return InviteDto(userId = invite.user?.id, eventId = invite.event?.eventId, state = invite.state)
        }
    }
}